akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -546,9 +549,8 @@ private[client] object NewHostConnectionPool {
 
           slot.debug("Establishing connection")
           val connection =
-            Source.fromGraph(requestOut.source)
-              .viaMat(connectionFlow)(Keep.right)
-              .to(responseIn.sink)
+            connectionFlow
+              .join(Flow.fromSinkAndSource(responseIn.sink, requestOut.source))
```

Comment:
How/why is this better?

Reply:
Good point, no good reason aside from two imports less :).

I just found that whole change in my big batch and didn't look closely enough. I find the previous version simpler to read so reverted.